src/main/java/Duke.java

```diff
@@ -1,10 +1,221 @@
+                try {
+                    String[] temp = input.split("todo "); //separates todo description
+                    String description = temp[1];
+                    ToDo todo = new ToDo(description);
+                    tasks.add(todo);
+                    Save.updateSaveFile(tasks);
+                    printTaskAddedStatement(tasks, todo);
+                } catch (IndexOutOfBoundsException exception) {
+                    printDottedLine();
+                    System.out.println("☹ OOPS!!! The description of a todo cannot be empty");
+                    printDottedLine();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else if (input.startsWith("deadline") & (tasks.size() <= maxIndex)) {
+                try {
+                    String[] temp = input.split("deadline | /by "); //separates deadline description and time
+                    String description = temp[1];
+                    String by = temp[2];
+                    Deadline deadline = new Deadline(description, by);
+                    tasks.add(deadline);
+                    Save.updateSaveFile(tasks);
+                    printTaskAddedStatement(tasks, deadline);
+                } catch (IndexOutOfBoundsException exception) {
+                    printDottedLine();
+                    System.out.println("☹ OOPS!!! The description of a deadline cannot be empty");
+                    printDottedLine();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else if (input.startsWith("event") & (tasks.size() <= maxIndex)) {
+                try {
+                    String[] temp = input.split(("event | /from | /to ")); //separates event description and times
+                    String description = temp[1];
+                    String from = temp[2];
+                    String to = temp[3];
+                    Event event = new Event(description, from, to);
+                    tasks.add(event);
+                    Save.updateSaveFile(tasks);
+                    printTaskAddedStatement(tasks, event);
+                } catch (IndexOutOfBoundsException exception) {
+                    printDottedLine();
+                    System.out.println("☹ OOPS!!! The description of a event cannot be empty");
+                    printDottedLine();
+                } catch (IOException e) {
+                    throw new RuntimeException(e);
+                }
+            } else if ((input.startsWith("todo") | input.startsWith("deadline") | input.startsWith("event"))
+                    & (tasks.size() > maxIndex)) {
+                printDottedLine();
+                System.out.println("☹ OOPS!!! The tasks list is currently full");
+                printDottedLine();
+            } else {
+                printUnknownCommandError();
+            }
+        }
+    }
```

Comment:
There is some deeply nested code in the method, and ample opportunities to improve SLAP.
Please consider refactoring the code.